tests/api/api_routes/workflow/test_advance.py

```diff
@@ -0,0 +1,176 @@
+    @patch(
+        "nestor_api.api.api_routes.workflow.advance.workflow_lib.advance_workflow", autospec=True
+    )
+    @patch("nestor_api.api.api_routes.workflow.advance.config_lib", autospec=True)
+    def test_advance_workflow(
+        self,
+        config_mock,
+        advance_workflow_mock,
+        non_blocking_clean_mock,
+        _configuration_mock,
+        _logger_mock,
+    ):
+        """Should properly return success response containing report."""
+        # Mock
+        config_mock.create_temporary_config_copy.return_value = "fake-path"
+        _configuration_mock.get_config_default_branch.return_value = "staging"
+        fake_config = {"workflow": ["master", "staging", "production"]}
+        config_mock.get_project_config.return_value = fake_config
+        advance_workflow_mock.return_value = (
+            WorkflowAdvanceStatus.SUCCESS,
+            [
+                {
+                    "name": "app-1",
+                    "tag": "0.0.0-sha-cf021d1",
+                    "step": "staging",
+                    "processes": [],
+                    "cron_jobs": [],
+                }
+            ],
+        )
+
+        # Tests
+        response = self.app_client.post("/api/workflow/progress/master")
+        data = response.get_json()
+
+        # Assertions
+        config_mock.create_temporary_config_copy.assert_called_once()
+        config_mock.change_environment.assert_called_once_with("staging", "fake-path")
+        advance_workflow_mock.assert_called_once_with("fake-path", fake_config, "master")
+        non_blocking_clean_mock.assert_called_once_with(
+            "fake-path", message_prefix="[/api/workflow/progress/<current_step>]"
+        )
+        self.assertEqual(response.status_code, HTTPStatus.OK)
+        self.assertEqual(
+            data,
+            {
+                "current_step": "master",
+                "message": "Workflow advance succeeded",
+                "report": [
+                    {
+                        "name": "app-1",
+                        "tag": "0.0.0-sha-cf021d1",
+                        "step": "staging",
+                        "processes": [],
+                        "cron_jobs": [],
+                    }
+                ],
+            },
+        )
+
+    @patch("nestor_api.api.api_routes.workflow.advance.Configuration", autospec=True)
+    @patch("nestor_api.api.api_routes.workflow.advance.non_blocking_clean", autospec=True)
+    @patch(
+        "nestor_api.api.api_routes.workflow.advance.workflow_lib.advance_workflow", autospec=True
+    )
+    @patch("nestor_api.api.api_routes.workflow.advance.config_lib", autospec=True)
+    def test_advance_workflow_with_status_failed(
+        self,
+        config_mock,
+        advance_workflow_mock,
+        non_blocking_clean_mock,
+        _configuration_mock,
+        _logger_mock,
+    ):
+        """Should properly return success response containing report."""
```

Comment:
Missed this one the first time, sorry
```suggestion
        """Should properly return failure response containing report."""
```